### PR TITLE
Fix projectiles collision

### LINKED ISSUE #413: [FEATURE] Projectile - Improve collision
When the projectile collides with a player, it explodes far away from him

### PR DESCRIPTION
**Pull request by manucamejo**
Closes #413 
- Improve projectile collision.
- Now you can shoot on the edges and the projectile doesn't explode.

### CODE REVIEW

File: client/Assets/Scripts/PlayerMovement.cs [hunk 1]
```diff
@@ -170,6 +170,23 @@ void UpdateProyectileActions()
                 projectiles.Add((int)gameProjectiles[i].Id, newProjectile);
             }
         }
+        // var toExplode = new List<int>();
```
Comment (tvillegas98) [inline client/Assets/Scripts/PlayerMovement.cs:173]: Remove this

File: client/Assets/Scripts/PlayerMovement.cs [hunk 2]
```diff
@@ -221,9 +204,26 @@ void UpdateProyectileActions()
                 movementDirection.Normalize();
 
                 Vector3 newPosition = projectile.transform.position + movementDirection * velocity * Time.deltaTime;
+                if (movementDirection.x > 0)
```
Comment (tvillegas98) [inline client/Assets/Scripts/PlayerMovement.cs:207]: This is not being used, can we remove it?

File: server/lib/dark_worlds_server/engine/runner.ex [hunk 3]
```diff
@@ -192,7 +192,8 @@ defmodule DarkWorldsServer.Engine.Runner do
     game_state = gen_server_state.server_game_state
 
     player_id = gen_server_state.current_players + 1
-    {:ok, new_game} = Game.spawn_player(game_state.game, player_id)
```
Comment (tvillegas98) [inline server/lib/dark_worlds_server/engine/runner.ex:195]: Restore this

Comment (tvillegas98) [reply server/lib/dark_worlds_server/engine/runner.ex:195]: And add an `Ok(game_2)` in `lib.rs:182`

File: server/native/gamestate/src/game.rs [hunk 4]
```diff
@@ -300,6 +300,73 @@ impl GameState {
...
+    ) -> Vec<u64> {
+        let mut affected_players: Vec<u64> = vec![];
+        players
+            .iter_mut()
+            .filter(|player| {
+                matches!(player.status, Status::ALIVE) && player.id != attacking_player_id
+            })
+            .for_each(|player| {
+                // TODO: Make the radius configurable
+                let radius = 10f64;
```
Comment (tvillegas98) [inline server/native/gamestate/src/game.rs:317]: I think lines between 317 and 328 doesn't need to be within the foreach loop

**Review by AminArria — APPROVED** [review]

**Review by tvillegas98 — APPROVED** [review]

**Pull request merged**